### PR TITLE
Update auto-update document

### PR DESCRIPTION
**Pull request by maruilian11**
for #9570

### CODE REVIEW

File: documents/autoupdate.md [hunk 1]
```diff
@@ -40,6 +40,7 @@ To add an `npm` hook to a library, update the `package.json` with configuration
 * `npmFileMap` is a list of files to take from the `npm` tarball and host on cdnjs
 * `basePath` will be ignored when copying over to the CDN
 * `files` is a pattern matcher allowing selection of multiple files
+  * you can use `**/*.!(...)` to exclude the files which are not needed. For example, you can use `**/*.!(flow|scss)` to exclude `*.flow` and `*.scss`.
```
Comment (PeterDaveHello) [inline documents/autoupdate.md]: This will not work.

Comment (maruilian11) [inline documents/autoupdate.md]: you mean `**/*.!(flow|scss)` will not work?

Comment (PeterDaveHello) [inline documents/autoupdate.md]: Does it work?

**Comment by maruilian11** [comment]
@PeterDaveHello I changed the rules, I'm not sure if they are too complex.

**Comment by PeterDaveHello** [comment]
Rules tested?

**Comment by maruilian11** [comment]
yes, they are tested.

File: documents/autoupdate.md [hunk 2]
```diff
@@ -40,7 +40,14 @@ To add an `npm` hook to a library, update the `package.json` with configuration
 * `npmFileMap` is a list of files to take from the `npm` tarball and host on cdnjs
 * `basePath` will be ignored when copying over to the CDN
 * `files` is a pattern matcher allowing selection of multiple files
+  * you can use `!(...)` to exclude the files which are not needed. For example, if you want to exclude `*.flow` and `*.scss`, you can use:
 
+  ```js
+  "files": [
+    "**/*/!(*.flow|*.scss)",  //to fetch all files under folders except *.flow and *.scss
+    "*.!(flow|scss)"  //to fetch all files in current path except *.flow and *.scss
```
Comment (PeterDaveHello) [inline documents/autoupdate.md]:
```json
    "**/*/!(*.flow|*.scss)",  // To fetch all files under all the folders except *.flow and *.scss
    "*.!(flow|scss)"          // To fetch all files in current path except *.flow and *.scss
```

**Comment by maruilian11** [comment]
@PeterDaveHello updated. Thanks!

**Comment by PeterDaveHello** [comment]
Let's add reference to `node-glob` doc about the glob usage.

**Comment by maruilian11** [comment]
@PeterDaveHello I added the reference.

**Comment by PeterDaveHello** [comment]
@kennynaoh please help verify the rules work 100% properly, thanks.

File: documents/autoupdate.md [hunk 3]
```diff
@@ -39,8 +39,15 @@ To add an `npm` hook to a library, update the `package.json` with configuration
...
 * `npmFileMap` is a list of files to take from the `npm` tarball and host on cdnjs
 * `basePath` will be ignored when copying over to the CDN
-* `files` is a pattern matcher allowing selection of multiple files
+* `files` is a pattern matcher allowing selection of multiple files, see [node-glob](https://github.com/isaacs/node-glob#glob-primer)
+  * you can use `!(...)` to exclude the files which are not needed. For example, if you want to exclude `*.flow` and `*.scss`, you can use:
 
+  ```js
+  "files": [
+    "**/*/!(*.flow|*.scss)",  //To fetch all files under all the folders except *.flow and *.scss
+    "*.!(flow|scss)"          //To fetch all files in current path except *.flow and *.scss
```
Comment (PeterDaveHello) [inline documents/autoupdate.md]: Can't these two example use a single style in the glob? `*.!(flow|scss)` or `!(*.flow|*.scss)`?

Comment (PeterDaveHello) [inline documents/autoupdate.md]: I don't think `"*.!(flow|scss)"` work anyway.

File: documents/autoupdate.md [hunk 4]
```diff
@@ -39,8 +39,14 @@ To add an `npm` hook to a library, update the `package.json` with configuration
 * `npmName` should map to the name of the library on `npm`
 * `npmFileMap` is a list of files to take from the `npm` tarball and host on cdnjs
 * `basePath` will be ignored when copying over to the CDN
-* `files` is a pattern matcher allowing selection of multiple files
+* `files` is a pattern matcher allowing selection of multiple files, see [node-glob](https://github.com/isaacs/node-glob#glob-primer)
+  * you can use `!(...)` to exclude the files which are not needed. For example, if you want to exclude `*.flow` and `*.scss`, you can use:
 
+  ```js
+  "files": [
+    "**/!(*.flow|*.scss)"  //To fetch all files except *.flow and *.scss
```
Comment (maruilian11) [inline documents/autoupdate.md]: @PeterDaveHello this rule has been tested.

Comment (kennynaoh) [inline documents/autoupdate.md]: @PeterDaveHello I think it's correct to exclude these two kinds of files under `basePath`

Comment (PeterDaveHello) [inline documents/autoupdate.md]: `"**/!(*.+(flow|scss))"`

**Comment by maruilian11** [comment]
@PeterDaveHello updated to `**/!(*.+(flow|scss))`.

File: documents/autoupdate.md [hunk 5]
```diff
@@ -39,8 +39,14 @@ To add an `npm` hook to a library, update the `package.json` with configuration
 * `npmName` should map to the name of the library on `npm`
 * `npmFileMap` is a list of files to take from the `npm` tarball and host on cdnjs
 * `basePath` will be ignored when copying over to the CDN
-* `files` is a pattern matcher allowing selection of multiple files
+* `files` is a pattern matcher allowing selection of multiple files, see [node-glob](https://github.com/isaacs/node-glob#glob-primer)
+  * you can use `!(...)` to exclude the files which are not needed. For example, if you want to exclude `*.flow` and `*.scss`, you can use:
 
+  ```js
+  "files": [
+    "**/!(*.+(flow|scss))"  //To fetch all files except *.flow and *.scss
```
Comment (PeterDaveHello) [inline documents/autoupdate.md]: Same problem again.
```js
   "**/!(*.+(flow|scss))"  // To fetch all files except *.flow and *.scss
```

Comment (PeterDaveHello) [inline documents/autoupdate.md]: What's the reason why to delete one another example here?

Comment (maruilian11) [inline documents/autoupdate.md]: I find `**/!(*.+(flow|scss))` can catch the same file as previous 2 examples did.

Comment (PeterDaveHello) [inline documents/autoupdate.md]: So all the libraries need this single super rule? We are writing examples for as the contributors' reference, I have no idea why you just touched the part without any discussion.

Comment (maruilian11) [inline documents/autoupdate.md]: sorry, it seems that I mistook the origin intention.
how about we provide two filemaps: 
```js
"files": [
  "**/*/!(*.+(flow|scss))",  // To fetch all files under all the folders except *.flow and *.scss 
  "!(*.+(flow|scss))"        // To fetch all files in current path except *.flow and *.scss
]
```
or you can use one single super rule to exclude the same files as above will do
```js
"files": [
  "**/!(*.+(flow|scss))"  // To fetch all files except *.flow and *.scss
]
```

Comment (maruilian11) [inline documents/autoupdate.md]: ping @PeterDaveHello

**Comment by PeterDaveHello** [comment]
Let's rebase and merge.

**Comment by maruilian11** [comment]
@PeterDaveHello I rebased the branch with the latest master branch.

**Pull request merged**